presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2068,18 +2098,34 @@ public void run(ExtendedHiveMetastore metastore)
                     // Not deleting the table may leave garbage behind. The former is much more dangerous than the latter.
                     // Therefore, the table is not considered added.
                 }
+
                 if (!done) {
                     throw e;
                 }
             }
+            tableCreated = true;
+        }
+
+        private boolean hasTheSameSchema(Table newTable, Table existingTable)
+        {
+            List<Column> newTableColumns = newTable.getDataColumns();
+            List<Column> existingTableColumns = existingTable.getDataColumns();
+            for (Column existingColumn : existingTableColumns) {
+                boolean found = newTableColumns.stream()
+                        .anyMatch(newColumn -> newColumn.getName().equals(existingColumn.getName()) && newColumn.getType().equals(existingColumn.getType()));
+                if (!found) {
```

Comment:
You can use `noneMatch()` to avoid the negation. Also, probably easier to read if you inline this

Reply:
```java
if (newTableColumns.stream().noneMatch(newColumn ->
        newColumn.getName().equals(existingColumn.getName()) &&
                newColumn.getType().equals(existingColumn.getType()))) {
    return;
}
```
(the extra indentation is unfortunate)